### PR TITLE
Allow multiple correct solutions for some checks of exercise 6 (Issue 44 fix)

### PR DESCRIPTION
**Pull request by babyygemperor**
**Shortly describe your proposed changes:**
 * https://github.com/N0W0RK/TUM_GAD_Tests_SS21/pull/45#pullrequestreview-676780843

**Link any issues if they have led to changes in the request:**
 * https://github.com/N0W0RK/TUM_GAD_Tests_SS21/issues/44

**Note the (sub-)exercises the tests are meant for:**

* GAD 06 - Verbessertes Sortieren

### CODE REVIEW

**Review by CrsiX — CHANGES_REQUESTED** [review]
Could you please address the code style issues mentioned by the GAD Worksheep? Especially the empty `if` and `else if` blocks  could be easily converted to one `if` statement with the `fail` call in it.

You don't need to address two style issues:
- max file length of more than 2000 lines
- max method length of more than 150 lines

Thanks

**Comment by N0W0RK** [comment]
I am reconfiguring our codestyle guidelines but we should work on reducing file and method length. We might build a testhelper class that automates giant tests like in this file here

**Comment by CrsiX** [comment]
Ideally, a good approach. But the problem might be: when you have a tool that generates test cases for something automatically, e.g. a function `f`, and also checks whether the results `f(x)` for certain `x` are correct, i.e. which provides `y=f(x)` _automatically_, how far are you away from building `f`? Because to accomplish this goal, you need to have some logic to know what's the right solution for a given `x`: you need to implement a way from `x` to `y`. And that's actually the students' tasks, so we **must not** do this.

**Comment by babyygemperor** [comment]
> for a given x: you need to implement a way from x to y. And that's actually the students' tasks, so we must not do this.

That was actually my idea behind all my test, I highly approve this message. The pull request should be removed.


Side note:
> how far are you away from building f?

Unfortunately in English we don't use the term building for the respective German word "_Abbildung_", instead we say, the range/codomain (The set of solutions) or in this case, simply the output (or the image if we want to sound fancy)

**Comment by CrsiX** [comment]
> The pull request should be removed.

Wait, you first approve my comment but then consider closing the PR. Why? Did I miss something here? :D

Yet another side note:

> Unfortunately in English we don't use the term building for the respective German word *"Abbildung"*, instead we say, the range/codomain (The set of solutions) or in this case, simply the output (or the image if we want to sound fancy)

Thanks for the explanation. My thoughts were actually more practically: 'building' refers to 'implementing' / 'constructing a function' :)

**Comment by babyygemperor** [comment]
> Wait, you first approve my comment but then consider closing the PR. Why?

Because sleep is very essential part of human life, although it takes 33% of our life time, ignoring it leads to illogical/half finished sentences, with the hope that the reader has telepathic abilities. 

> My thoughts were actually more practically: 'building' refers to 'implementing' / 'constructing a function' :)

Ah okay, well in that case, "implementing"/"constructing a function" sounds more complete than "building". Building just sounded off in that sentence, hence my 2 cents

**Comment by N0W0RK** [comment]
I have created #48 to further explaion and thack the progress of my proposal.

I am far from suggesting that we implement the actual solutions in our test. I rather want to streamline the process of having multiple array checks into a more clear and simple setup.

This PR imo does not need to be closed but simply fix the codestyle issue

File: GAD 6 - Verbessertes Sortieren/tests/simplesort/DualPivotFinderTester.java
Comment (N0W0RK) [inline GAD 6 - Verbessertes Sortieren/tests/simplesort/DualPivotFinderTester.java:287]: This test also fits my issue but has not been fixed. There are a lot of lines like this

**Review by N0W0RK — CHANGES_REQUESTED** [review]

**Comment by babyygemperor** [comment]
Done

**Comment by babyygemperor** [comment]
All the issues have been fixed here since 5 days as well, I'm still waiting for someone to approve it.

**Comment by N0W0RK** [comment]
I will merge if you resolve all codestyle issues associated with your changes

**Comment by babyygemperor** [comment]
Done

**Comment by CrsiX** [comment]
Thanks for addressing the issues lined out by us and the style checker. I will look into them again later :)

**Review by N0W0RK — APPROVED** [review]

**Review by CrsiX — APPROVED** [review]

**Pull request merged**
